sandbox/anvi-script-filter-hmm-hit-table

```diff
@@ -0,0 +1,223 @@
+#!/usr/bin/env python
+# -*- coding: utf-8
+
+import sys
+import os
+import numpy as np
+import pandas as pd
+
+import anvio
+import anvio.data.hmm
+import anvio.db as db
+import anvio.hmmops as hmmops
+import anvio.bamops as bamops
+import anvio.terminal as terminal
+import anvio.filesnpaths as filesnpaths
+
+from anvio.dbops import ContigsDatabase
+from anvio.parsers import parser_modules
+from anvio.argparse import ArgumentParser
+from anvio.tables.hmmhits import TablesForHMMHits
+from anvio.errors import ConfigError, FilesNPathsError
+
+__author__ = "Developers of anvi'o (see AUTHORS.txt)"
+__copyright__ = "Copyleft 2015-2020, the Meren Lab (http://merenlab.org/)"
+__credits__ = []
+__license__ = "GPL 3.0"
+__version__ = anvio.__version__
+__maintainer__ = "Matthew S. Schechter"
+__email__ = "mschechter@uchicago.edu"
+__provides__ = ["contigs-db", "hmm-hits"]
+__requires__ = ["contigs-db","hmm-source", "hmm-hits"]
+__description__ = ("Add or remove entries in a contigDB hmm_hits table.")
+
+pp = terminal.pretty_print
+
+class FilterHmmHitsTable(object):
+    """A class to filter hmm_hits from domtblout from hmmsearch."""
+    def __init__(self, args, quiet=False, run=terminal.Run(), progress=terminal.Progress()):
+        self.args = args
+        self.run = run
+        self.progress = progress
+
+        A = lambda x: args.__dict__[x] if x in args.__dict__ else None
+        self.contigs_db_path=A("contigs_db")
+        self.domtblout=A("domtblout")
+        self.hmm_source=A("hmm_source")
+        self.target_coverage=A("target_coverage")
+        self.query_coverage=A("query_coverage")
+        self.list_hmm_sources=A("list_hmm_sources")
+
+        if self.list_hmm_sources:
+            ContigsDatabase(self.contigs_db_path).list_available_hmm_sources()
+            sys.exit()
+
+        self.sanity_checks()
+
+        self.import_domtblout()
+
+        x = self.filter_domtblout()
+
+        self.get_hmm_hit_table(hmmsearch_tbl=x)
+
+        # remove the tmp file
+        os.remove(x)
```

Comment:
Usually when I remove files, I include the option to keep them around if the user runs with `--debug`. This way, you don't clutter up the tmp directory on their computer, but the user can explicitly request to see this file in case something goes wrong and they need to check it.

Reply:
Great idea, I added that funcitonality here: [9209511](https://github.com/merenlab/anvio/pull/1678/commits/9209511bd36c8279c131548ed9d36aa302d0f250)